include/boost/beast/http/impl/write.hpp

```diff
@@ -227,6 +237,8 @@ class write_op
                         s_, sr_, std::move(*this));
                 }
                 bytes_transferred_ += bytes_transferred;
+                if (!ec && st_.cancelled() != net::cancellation_type::none)
+                    ec = net::error::operation_aborted;
```

Comment:
I suspect this should not cancel when cancellation type is total.
does the op need to apply a filter to the cancellation state to prevent total cancellation (since it can't be supported)?
partial would be ok
terminal also ok

Reply:
It's `terminal` by filter in the state already.

https://github.com/chriskohlhoff/asio/blob/fc901d79c11c72a0ba81764820d3c6a788c0f075/asio/include/asio/cancellation_state.hpp#L88